contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -209,6 +229,18 @@ where
 				// We need to reject transaction amounts that are larger than u128 since we expect
 				// the balance type of the runtime to be a u128; For a U256 to be converted to a
 				// u128 without truncating, the last two words should be zero
+				let msg = Ics20TransferMsg {
```

Comment:
do we need to clone? can't we just operate based on references?

Reply:
yes and not. until we get generics over lifetime https://blog.rust-lang.org/inside-rust/2022/07/27/keyword-generics.html, could do 4 things, path each filed separately OR move into struct and move out OR make special referene field stuct OR refactor lower layers to work with struct. 

given these options and hard to decide desing imho, better leave as is with smallest amount of code.

Reply:
so in this case best solution is clone. anyway most of cenauri is very clone heavy. so until it all fixed, one small piece is not essential.